File: src/ArchiveNet.Domain/Artist.cs
namespace ArchiveNet.Domain;

public record Artist(int Id, Name Name, NameCollection AlsoKnownAs)
{
	public Artist()
		: this(default, new Name(""), new NameCollection() )
	{
		
	}
	public Artist(Name name)
		: this(default, name, new NameCollection())
	{
		
	}

	public Artist(int id, Name name)
		: this(id, name, new NameCollection())
	{
		
	}

	public Artist(int id)
		: this(id, new Name(string.Empty), new NameCollection())
	{
		
	}
}

File: src/ArchiveNet.Web.Gui/src/app/dto/Artist.ts
import { EmptyName, Name } from "./Name";
import { EmptyNameCollection, NameCollection } from "./NameCollection";

export class Artist {
	id: number = 0;
	name: Name = new Name('');
	alsoKnownAs: NameCollection = new NameCollection([]);

	constructor(id: number, name: Name, alsoKnownAs: NameCollection) {
		this.id = id;
		this.name = name;
		this.alsoKnownAs = alsoKnownAs;
	}
}

export class EmptyArtist extends Artist {
	constructor() {
		super(0, new EmptyName(), new EmptyNameCollection());
	}
}